portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,19 +27,44 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-    ArrayList<String> user = new ArrayList<>(
-            Arrays.asList("Roland","Naijuka","03/20","University of Rochester","Computer Science")
-        );
+    private final int MAX_COMMENTS = 3;
+    private ArrayList<Comments> userComments = new ArrayList<>();
+    
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        String userInJson = convertToJson(user);
-        response.setContentType("application/json;");
-        response.getWriter().println(userInJson);
+        response.setContentType("application/json");
+
+        //Convert the arraylist to json string
+        String comments = new Gson().toJson(userComments);
+        response.getWriter().println(comments);
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String comment = getParameter(request,"comment","");
+        String username = getParameter(request, "name","");
+        if(comment.length() != 0){
+            //check if user sent their name
+            Comments newComment = username.length() == 0 ? new Comments(comment) : new Comments(comment, username);
+            if(userComments.size() < MAX_COMMENTS){
+                userComments.add(newComment);
+            }
+            else{
+                userComments.clear();
+                userComments.add(newComment);
+            }
```

Comment:
looks like you call userComments.add(newComment) either way, the only difference is that the previous comments might need to be cleared first 
I would restructure it like this to avoid repeating code and shorten it a bit: 
if (userComments.size() >= MAX_COMMENTS) {
  userComments.clear();
}
userComments.add(newComments);

Reply:
> looks like you call userComments.add(newComment) either way, the only difference is that the previous comments might need to be cleared first
> I would restructure it like this to avoid repeating code and shorten it a bit:
> if (userComments.size() >= MAX_COMMENTS) {
> userComments.clear();
> }
> userComments.add(newComments);

Thank you for this comment. I like this way.